docs: Fix minor issues in docs/usage/applications.rst (#3114)
litestar-org/litestar: docs/usage/applications.rst

    * :ref:`Routing - Registering Routes <usage/routing/overview:Registering Routes>`

> .. _startup-and-shutdown:

Reviewer: I thought that with [`autosectionlabel`](https://www.sphinx-doc.org/en/master/usage/extensions/autosectionlabel.html#module-sphinx.ext.autosectionlabel) we didn't need to do things like this

Author: Updated to use section reference.

Only did a quick research on rst yesterday and didnt see that as an option, but it's cleaner.

Author: That didnt work out. Does it need a full path? Like `<usage/application:Startup and Shutdown>`?

Reviewer 2: You just need to get rid of the `:ref:` role, then it should work.

Using Application State
-----------------------

> As seen in the examples for the `on_startup / on_shutdown <#startup-and-shutdown>`_ ,

Reviewer: ```suggestion
As seen in the examples for the `on_startup / on_shutdown <#startup-and-shutdown>`_,
```

> callables passed to these hooks can receive an optional kwarg called ``app``, through which is the application's state object

Reviewer 2: ```suggestion
callables passed to these hooks can receive an optional kwarg called ``app``, through which the application's state object
```

> callables passed to these hooks can receive an optional kwarg called ``app``, through which is the application's state object
> and other properites can be accessed.

Reviewer 2: ```suggestion
callables passed to these hooks can receive an optional kwarg called ``app``, through which the application's state object
and other properties can be accessed.
```
